Fitter/src/net/relinc/fitter/GUI/HomeController.java

```diff
@@ -209,6 +209,36 @@ public void handle(MouseEvent event) {
 		
 		
 		
+	}
+
+	public void setFitterFromTrimSelection(Double xAxisBegin, Double xAxisEnd){
+		FitableDataset theData = datasetsListView.getSelectionModel().getSelectedItem();
+		theData.setBeginFromXValue(xAxisBegin);
+		theData.setEndFromXValue(xAxisEnd);
+		renderCharts();
+
+		FitableDataset f = getCurrentDataset();
+		if(f == null)
+			return;
+		int fit = (int)polynomialOrderScrollBar.getValue();
+		f.setPolynomialFit(fit);
+		f.renderFittedData();
+		renderCharts();
+
+		double maxY = -Double.MAX_VALUE;
+		double minY = Double.MAX_VALUE;
+		for(int i = 0; i < theData.origX.size(); i++){
+			if( theData.origX.get(i) >= xAxisBegin && theData.origX.get(i) <= xAxisEnd) {
+				if( theData.origY.get(i) > maxY ) { maxY = theData.origY.get(i); }
+				if( theData.origY.get(i) < minY ) { minY = theData.origY.get(i); }
+			}
+		}
+
+		beginRectangle = new Point2D(xAxisBegin - (xAxisEnd-xAxisBegin)*(0.05), maxY + (maxY-minY)*(0.1));
+		endRectangle = new Point2D(xAxisEnd + (xAxisEnd-xAxisBegin)*(0.05), minY-(maxY-minY)*(0.1));
+		setChartBounds();
+		fitChartYAxis.setTickUnit((maxY-minY)*0.25);
+		renderCharts();
```

Comment:
You call `renderCharts()` 3 times in this method, but it only should get called once I think. 

The UI updates only after you yield control back to the UI thread. So calling `renderCharts` does not cause the charts to immediately update, it's only after all the code runs that it updates. So calling it multiple times has no visible effect and will just be slower